src/Tribe/Repository/Core_Read_Interface.php

```diff
@@ -109,13 +113,14 @@ public function found();
 	 *
 	 * @since 4.1.3
 	 * @since 5.2.0 Added the `$return_generator` and `$batch_size` parameters.
+	 * @since TBD   Updated `return` to more accurately reflect what can be returned.
 	 *
 	 * @param bool $return_generator Whether to return a generator of post IDs instead of an array of post IDs.
 	 * @param int  $batch_size       The number of post IDs to fetch at a time when using a generator; ignored
 	 *                               if `$return_generator` is false.
 	 *
-	 * @return array<int>|Generator<int> An array of all the matching post IDs, or a generator of them
-	 *                                   if `$return_generator` is true.
+	 * @return array<int|WP_Post|object>|\Generator<int|WP_Post|object> An array of all the matching post IDs, or a generator of them
```

Comment:
```suggestion
	 * @return array<int|WP_Post|object>|Generator<int|WP_Post|object> An array of all the matching post IDs, or a generator of them
```

import at top with

`use Generator;`